File: leet/prefix_sum/find_pivot_index_724.java
class find_pivot_index_724 {
    public int pivotIndex(int[] nums) {
        int right = 0;
        int left = 0;
        for (int i : nums) {
            right += i;
        }
        for (int i = 0; i < nums.length; i++) {
            left += (i != 0) ? nums[i - 1] : 0;
            right -= nums[i];
            if (left == right) {
                return i;
            }
        }
        return -1;
    }
}
// 1ms

File: leet/prefix_sum/find_pivot_index_724.py
from typing import List

class Solution:
    # def pivotIndex(self, nums: List[int]) -> int:
    #     for i in range(len(nums)):
    #         left = sum(nums[:i])
    #         right = sum(nums[i+1:])
            
    #         if left == right:
    #             return i
    #     return -1
    # --------------------------------------------------------------
    # ! cleaver 只要一次sum()就解決
    def pivotIndex(self, nums: List[int]) -> int:
        right = sum(nums)
        left = 0
        for i in range(len(nums)):
            left += nums[i-1] if i != 0 else 0
            right -= nums[i]
            if left == right:
                return i
        return -1

aa = Solution()
print(aa.pivotIndex([0,1,2,3,4,5]))